### ISSUE TITLE
esy bootstrap Error on Windows

### ISSUE THREAD

**Issue by pianocomposer321** [onivim/oni2#1682]
Here's the error:

```
info building @opam/ocaml-lsp-server@github:ocaml/ocaml-lsp:ocaml-lsp-server.opam#04733ed@d41d8cd9
info building reason-jsonrpc@1.5.0@d41d8cd9
error: build failed with exit code: 1
  build log:
    # esy-build-package: building: @opam/ocaml-lsp-server@github:ocaml/ocaml-lsp:ocaml-lsp-server.opam#04733ed
    # esy-build-package: pwd: C:\Users\Tad\.esy\3_\b\opam__s__ocaml_lsp_server-ed59ac86
    # esy-build-package: running: "dune" "build" "-j" "4" "--root" "." "--ignore-promoted-rules" "--no-config" "--profile" "release" "ocaml-lsp-server.install"
    File "ocaml-lsp-server/src/dune", line 8, characters 3-11:
    8 |    cmdliner))
           ^^^^^^^^
    Error: Library "cmdliner" not found.
    Hint: try:
    dune external-lib-deps --missing --no-config --root . --ignore-promoted-rules
    --profile release -j 4 ocaml-lsp-server.install
    error: command failed: "dune" "build" "-j" "4" "--root" "." "--ignore-promoted-rules" "--no-config" "--profile" "release" "ocaml-lsp-server.install" (exited with 1)
    esy-build-package: exiting with errors above...

  building @opam/ocaml-lsp-server@github:ocaml/ocaml-lsp:ocaml-lsp-server.opam#04733ed
esy: exiting due to errors above
```

**Comment by zbaylin** [comment]
Strangely enough I was having this exact same issue on Linux (Ubuntu 19.10). Doesn't seem to just be a windows issue -- which makes sense. If I can find a fix for it I will let you know @pianocomposer321!

**Comment by tylerperkins** [comment]
I'm also seeing this on MacOS 10.15.5.

**Comment by pianocomposer321** [comment]
@bryphe Any idea what's causing the error?

**Comment by bryphe** [comment]
@pianocomposer321 - not sure - fortunately (or unfortunately, for reproducing) - this isn't showing up for me locally, or on our CI. It could be a stale artifact in your build cache.

Some things to try:
- Make sure you have the latest `esy` installed (`esy --version` should be `0.6.6`)
- Try cleaning out the build folder, in case there are old artifacts (delete and re-clone the repo)
- Delete the `esy` cache in case there are some stale artifacts - `rm -rf ~/.esy`

It looks like the error is with `@opam/ocaml-lsp-server@github:ocaml/ocaml-lsp:ocaml-lsp-server.opam` too, which isn't actually needed for running Onivim - it's a development time dependency. You could try removing this line in the `package.json` as well: https://github.com/onivim/oni2/blob/15885387494e6036e815f4570db0efee2864da00/package.json#L272

**Comment by pianocomposer321** [comment]
Thanks @bryphe I will try those things later!

**Comment by tylerperkins** [comment]
Indeed, I did have an old version of esy (0.5.8). Doing `npm update` did not update it, so I uninstalled it and installed again. Now I see this:

    $ esy --version
    Failed to execute process '/usr/local/bin/esy'. Reason:
    exec: Exec format error
    The file '/usr/local/bin/esy' is marked as an executable but could not be run by the operating system.

    $ npm info esy

    esy@0.6.6 | MIT | deps: 1 | versions: 135
    Package builder for esy.

    bin: esy
    ...

And note this:

    $ ls -l /usr/local/lib/node_modules/esy/_build/default/bin/
    total 0
    -rwxr-xr-x  1 tyler  staff  0 Oct 26  1985 esy.exe*

See how the size is 0, and it's named in Windows-style, with the `.exe`? Am I getting the right version for MacOS?

**Comment by CrossR** [comment]
The `exe` shouldn't be a cause for concern, a lot of the OCaml / Reason stuff has that to signify its executable, even if its a bit confusing when on mac or Linux.

It looks like the `esy` install is a bit broken though...if you can't get it working, you could probably drop a message either in the "native-development" channel in the ReasonML discord, or the "esy" channel in the OCaml discord and one of the people who work on `esy` may have a better idea. It looks like the install didn't work correctly though...so maybe a second uninstall/reinstall is needed.

**Comment by tylerperkins** [comment]
OK, @CrossR. Thanks for helping an esy / ReasonML newby. BTW, reinstalling again didn't help.
